src/main/java/com/cedarsoftware/util/io/WriteOptionsBuilder.java

```diff
@@ -0,0 +1,817 @@
+        return classLoader;
+    }
+
+    /**
+     * @param classLoader ClassLoader to use when writing JSON to resolve String named classes.
+     * @return WriteOptionsBuilder for chained access.
+     */
+    public WriteOptionsBuilder classLoader(ClassLoader classLoader) {
+        this.classLoader = classLoader;
+        return this;
+    }
+
+    /**
+     * @return boolean true if showing short meta-keys (@i instead of @id, @ instead of @ref, @t
+     * instead of @type, @k instead of @keys, @v instead of @values), false for full size. 'false' is the default.
+     */
+    public boolean isShortMetaKeys() {
+        return shortMetaKeys;
+    }
+
+    /**
+     * @param shortMetaKeys boolean true to turn on short meta-keys, false for long.
+     * @return WriteOptionsBuilder for chained access.
+     */
+    public WriteOptionsBuilder shortMetaKeys(boolean shortMetaKeys) {
+        this.shortMetaKeys = shortMetaKeys;
+        return this;
+    }
+
+    /**
+     * @param aliasTypeNames Map containing String class names to alias names.  The passed in Map will
+     *                       be copied, and be the new baseline settings.
+     * @return WriteOptionsBuilder for chained access.
+     */
+    public WriteOptionsBuilder aliasTypeNames(Map<String, String> aliasTypeNames) {
+        aliasTypeNames.forEach(this::addUniqueAlias);
+        return this;
+    }
+
+    /**
+     * Aliases the fully qualifed class name to its short name
+     *
+     * @param type Class to alias
+     * @return ReadOptions for chained access.
+     */
+    public WriteOptionsBuilder aliasTypeName(Class<?> type) {
+        return aliasTypeName(type.getName(), type.getSimpleName());
+    }
+
+
+    /**
+     * @param type  Class to alias
+     * @param alias String shorter name to use, typically.
+     * @return ReadOptions for chained access.
+     */
+    public WriteOptionsBuilder aliasTypeName(Class<?> type, String alias) {
+        aliasTypeNames.put(type.getName(), alias);
+        return this;
+    }
+
+    /**
+     * @param typeName String class name
+     * @param alias    String shorter name to use, typically.
+     * @return WriteOptionsBuilder for chained access.
+     */
+    public WriteOptionsBuilder aliasTypeName(String typeName, String alias) {
+        addUniqueAlias(typeName, alias);
+        return this;
+    }
+
+    /**
+     * Since we are swapping keys/values, we must check for duplicate values (which are now keys).
+     * @param typeName String fully qualified class name.
+     * @param alias String shorter alias name.
+     */
+    private void addUniqueAlias(String typeName, String alias) {
+        Convention.throwIfClassNotFound(typeName, getClassLoader());
+        Convention.throwIfKeyExists(aliasTypeNames, typeName, "Tried to create @type alias" + typeName + " -> " + alias + ", but it is already aliased to: " + aliasTypeNames.get(typeName));
+
+        aliasTypeNames.put(typeName, alias);
```

Comment:
Both sides need to be checked.  The key must not be a duplicate.  The value must also be unique.  The Alias Map is really a "Bimap."  Since we don't have a Gauva, we need to check both keys and values for duplicates.  Keys is O(1) and values check will be O(n) but this is at start-up, so it doesn't really matter.  Without this check, I could alias java.util.Synchronized$Collection to "syncCol" and then also map "ConcurrentSkipListSet" to "syncCol".  It will pass the key check, but since values are not being checked for uniqueness, it would be accepted.  Then when we go to unalias, we would be only working half-right/half-wrong.